src/docs/transaction-clustering/index.mdx

```diff
@@ -0,0 +1,147 @@
+
+However, in some cases the SDK does not have enough information to pick a meaningful group
+like the name of a view or a route of a web transaction,
+and it has to fall back to the raw URL (or rather,
+[its path component](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax)).
+
+First and foremost, this makes it harder for the user to extract insights from [Performance
+metrics](https://docs.sentry.io/product/performance/metrics/), because instead of presenting averages, percentiles and distributions of groups of transactions that logically
+belong together, we end up with a bunch of one-off transaction groups.
+
+```
+/account/settings/nf215j634705291cvcm725757m53451365023487cv562n905n2634590xn26340/
+/issues/1234567890
+/saml/acs/my-organization/
+/settings/e1fb593b4c3a4d388e2f6ea8c92c3abc/
+```
+
+Secondly, ingesting a high number of one-off transaction names puts a cost on our infrastructure,
+because with [metrics](https://getsentry.github.io/relay/relay_metrics/index.html), the cost of storage is determined by _cardinality_ (the number of distinct groups), not volume (the number of events ingested).
+
+### << unparameterized >>
+
+An intermediate solution to the cost problem was to drop the transaction tag from Performance metrics
+when the transaction name is a raw URL. This effectively groups _all_ URL transactions into one big group,
+which also creates a bad user experience:
+
+![Project with only unparameterized transactions](./unparameterized.png)
+
+## The Solution
+
+### Pattern-based Identifier Scrubbing
+
+In a first step, Relay strips common identifiers such as UUIDs, integer IDs and hashes
+from URL transactions ([code](https://github.com/getsentry/relay/blob/e543401086c556b769b67ccb96fbfaca411696c8/relay-general/src/store/transactions/processor.rs#L273-L277)).
+For example,
+
+```python
+"/hash/4c79f60c11214eb38604f4ae0781bfb2/diff"
+# becomes
+"/hash/*/diff"
+```
+
+### Automatic Transaction Clustering
+
+Some identifiers cannot be detected by looking at a single transaction. For example,
+free-form user names cannot be dinstinghuished from low-cardinality parts of a URL.
+
+To detect high-cardinality segments in the URL, we accumulate a set of unique URL transaction names for each project, split them by `"/"`, and build a tree from its path segments (similar to a [trie](https://en.wikipedia.org/wiki/Trie) data structure):
+
+![tree of URL path segments](./tree.png)
+
+If the number of children of a node surpass a threshold (currently set to 200), we treat the path segment as high-cardinality, and fold its children into a single node:
+
+![tree with merged segments](./merged.png)
+
+For every merged segment, we create a glob-like replacement rule that can be used to scrub identifiers from new transactions. In the example above, the rule would be
+
+```
+/user/*/**
+```
+
+If a transaction name matches this glob, the segment matching `*` gets replaced, thereby erasing the identifier.
```

Comment:
I'd use "corresponding" (or another word) over "matching", since technically every segment "matches" `*`.
```suggestion
If a transaction name matches this glob, the segment corresponding to `*` get replaced, thereby erasing the identifier.
```